pkg/cloudprovider/providers/openstack/openstack_instances.go

```diff
@@ -134,11 +150,7 @@ func (i *Instances) NodeAddresses(name string) ([]api.NodeAddress, error) {
 
 // ExternalID returns the cloud provider ID of the specified instance (deprecated).
 func (i *Instances) ExternalID(name string) (string, error) {
-	srv, err := getServerByName(i.compute, name)
-	if err != nil {
-		return "", err
-	}
-	return srv.ID, nil
+	return probeInstanceID(i.compute, name)
```

Comment:
Now that I look closer at this, I think we should just pass `os.localInstanceID` into the `Instances` object (ie: add it to the struct up the top and set it in `Instances()`).  Then this just becomes:
```go
return i.localInstanceID, nil
```

(and similar in `InstanceID` below)

In particular, the local instance ID never changes, so we should be caching it aggressively.  And we're already doing that (and indeed aborting if we can't find it) way back in `newOpenStack()`.  (Sorry for not suggesting this easier version earlier)